xwiki-commons-core/xwiki-commons-diff/xwiki-commons-diff-display/src/main/java/org/xwiki/diff/display/UnifiedDiffDisplayer.java

```diff
@@ -140,4 +141,26 @@ default <E, F> List<UnifiedDiffBlock<E, F>> display(DiffResult<E> diffResult, Li
     {
         return display(diffResult, config);
     }
+
+    /**
+     * Allows to retrieve a diff block containing a specific line to be retrieved from an existing diff.
+     * The idea of this method is to compute a specific contextual diff block, based on an already existing list of
+     * diff blocks: this contextual diff block only contains the line to be found and the surrounding lines matching the
+     * context line number given in {@link UnifiedDiffConfiguration#getContextSize()}.
+     *
+     * @param <E> the type of elements that were compared to produce the diff
+     * @param <F> the type of sub-elements that can be compared to produce an in-line diff when an element is modified
+     * @param originalDiff the original list of unified diff blocks computed from
+     *                     {@link #display(DiffResult, UnifiedDiffConfiguration)}.
+     * @param previousLineNumber the index of the line to be retrieved in case of deletion or context change, in case of
+     *                           addition the value should be negative
+     * @param nextLineNumber the index of the line  to be retrieved in case of addition or context change, in case of
+     *                          deletion the value should be negative
+     * @param configuration the same configuration used to compute the original diff
+     * @return a single diff block whose size is based on the selected line and on the size of the context
+     * @since 15.0RC1
+     */
+    @Unstable
+    <E, F> UnifiedDiffBlock<E, F> getContextDiffBlock(List<UnifiedDiffBlock<E, F>>
```

Comment:
I'm not sure ``UnifiedDiffDisplayer`` is the right place for this method. Judging from the input parameters, this method is called on the output produced by the unified diff displayer, so I think it make more sense to have it in a separate component. It's not easy to find a good name, maybe something like ``UnifiedDiffBlockExtractor``.